### PR TITLE
Add phpstan disallowed calls extension

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR adds the disallowed calls for phpstan, so we can forbid usage of deprecated namespaces, migrated functions, etc. This also removes the `prestashop/phpstan-prestashop` as the package is not used (removed in 2021 (857ee5f54f7c505cdc9d381977f1052ab9b419ea) and incompatible with the phpstan extension installer
| Type?             | improvement 
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Related PRs       | If theme, autoupgrade or other module change is needed, provide a link to related PRs here.
| How to test?      | This is tooling for CI, no tests needed by @PrestaShop/qa-team.

I introduce this PR as we can use this tool to prevent usages of `TranslatorInterface` that is deprecated and will be removed to ease migration.

### CODE REVIEW

**Review by nicosomb — DISMISSED** [review]
👍

**Review by zuk3975 — DISMISSED** [review]

**Review by zuk3975 — DISMISSED** [review]

**Review by nicosomb — DISMISSED** [review]

**Review by nicosomb — DISMISSED** [review]

**Review by zuk3975 — DISMISSED** [review]

**Comment by HanaRebaiQA** [comment]
Hello @FabienPapet 

This PR should be ready for testing when the CI is all passed.
image

So i'll remove the Waiting for QA label.

Thanks!

**Review by jolelievre — DISMISSED** [review]

**Comment by jolelievre** [comment]
Hi @HanaRebaiQA 

actually, this PR is only about static code analyze and used for the CI no functional or even code impact, it's a pre-requisite for another PR that will rely on a new rule for PHStan (the package for this rule is installed in this PR).

So no QA needed at all, we just need to wait for the CI, that's why I removed the Waiting for QA label

**Comment by HanaRebaiQA** [comment]
Hello @jolelievre 

Thank You for your feedback! 
Even, do not need automated tests on it?

**Comment by FabienPapet** [comment]
They're not needed as this only fixes development environment

**Comment by HanaRebaiQA** [comment]
It's OK for us then!

Thank You

**Comment by FabienPapet** [comment]
Ci failed because of the order of the `config.allow-plugins` 😑  need an approval again @nicosomb @zuk3975 

cf https://github.com/PrestaShop/PrestaShop/actions/runs/3629709814/jobs/6122232812

**Review by nicosomb — APPROVED** [review]

**Review by zuk3975 — APPROVED** [review]

**Pull request merged**
